docs/spec/BeakerSpec.tex

```diff
@@ -0,0 +1,994 @@
+      procedure reverted. \tabularnewline
+    SYSCALL\_FAIL    & 66 & System call specific error data & The system call
+      failed for reasons specific to the system call. \tabularnewline
+    SYSCALL\_NOEXIST & 111 & None & The syscall integer values is not a valid
+      syscall as defined in Table \ref{syscall-numbers} \tabularnewline
+    \hline
+  \end{tabular}
+\end{table}
+
+\subsection{Capability Subsets}\label{cap-subsets}
+
+Capability $B$ is a subset\footnote{$A$ subset, not a \emph{strict} or
+\emph{proper} subset.} of Capability $A$, if everything that can be done using
+Capability $B$ can be done using Capabilty $A$. The following rules apply:
+
+\begin{itemize}
+  \item $A$ is always a subset of itself. $A \subseteq A$
+  \item Transitivity: $A \subseteq B \land B \subseteq C \implies A
+  \subseteq C$
+\end{itemize}
+
+As the nature of each capability is different, the way subsets are defined is
+different for each one. The following sections will define how subsets are
+defined for each capability.
+
+\subsection{Call System Call}
+
+\subsubsection{System Call Format}
+Type Value: 3
+
+Data: Note that here the offset is from after system call type.
+
+Note that here we could pack the data much more densely, but we don't in order
+to err on the side of simplicity and stick as close as we can to 32-byte values.
+
+\begin{table}[H]
+  \caption{Call procedure system call format.}
+  \centering{}%
+  \begin{tabular}{l|p{0.7\textwidth}}
+    \hline
+    Byte Offset & Description\tabularnewline
+    \hline
+    \hline
+    0x0 & Capability index; 32 bytes \tabularnewline
+    0x20 & Prefix size, 1 bytes \tabularnewline
+    0x21 & 8 unused bytes \tabularnewline
+    0x28 & Procedure key, 24 bytes \tabularnewline
+    \hline
+    End &  \tabularnewline
+    \hline
+  \end{tabular}
+\end{table}
+
+\subsubsection{Capability Format}
+The capability format for the Call Procedure system call defines the a range of
+procedure keys what the capability allows one to call. This is defined as a base
+procedure key $b$ and a prefix $s$. Given this capability, a procedure may call
+any procedure where the first $s$ bits of the key of that procedure key are the
+same as the first $s$ bits of procedure key $b$.
+
+The values of this capability are packed into a single 32-byte value.
+
+\begin{table}[H]
+  \caption{Call procedure capability format.}
+  \centering{}%
+  \begin{tabular}{l|l|p{0.5\textwidth}}
+    \hline
+    Key Offset & Value & Description\tabularnewline
+    \hline
+    \hline
+    0x00 & -    & The prefix and base procedure key, the first byte is the prefix
+                 size which is in the interval [0,24], bytes [0x1,0x7] are
+                 unused and undefined, bytes [0x8,0x32] are the 24 bytes of the
+                 base procedure key. \tabularnewline
+    \hline
+    End &  \tabularnewline
+    \hline
+  \end{tabular}
+\end{table}
+
+\includegraphics[width=\textwidth]{media/ProcedureKeyBytes.pdf}
```

Comment:
Broken Link

Reply:
This is produced via the makefile from an SVG. Since I'm committing the PDF I won't commit these too unless you want me too, in which case I will.